main/app-menu.js

```diff
@@ -0,0 +1,56 @@
+const { Menu, MenuItem, ipcMain } = require('electron')
+const { ipcMainEvents } = require('./ipc')
+
+function setupAppMenu (/** @type {import('./typings').Context} */ ctx) {
+  if (process.platform !== 'darwin') return
+
+  // Add "Check for updates..." item to the Application menu on MacOS
+  const menu = Menu.getApplicationMenu()
+  if (!menu) return
+
+  menu.items[0].submenu?.insert(1, new MenuItem({
+    id: 'checkForUpdates',
+    label: 'Check For Updates...',
+    click: () => { ctx.manualCheckForUpdates() }
+  }))
+  menu.items[0].submenu?.insert(2, new MenuItem({
+    id: 'checkingForUpdates',
+    label: 'Checking For Updates',
+    enabled: false,
+    visible: false
+  }))
+
+  Menu.setApplicationMenu(menu)
+  setupIpcEventListeners(menu)
+}
+
+/**
+ * @param {Electron.Menu} appMenu
+ */
+function setupIpcEventListeners (appMenu) {
+  ipcMain.on(ipcMainEvents.UPDATE_CHECK_STARTED, () => {
+    getItemById('checkForUpdates').enabled = false
```

Comment:
```suggestion
```

I don't see the purpose of this, afaic if an item is invisible it's also disabled

Reply:
Same question applies to the tray configuration

Reply:
To be honest, I am not sure about this either. I have simply copied what I found in ipfs-desktop:

https://github.com/ipfs/ipfs-desktop/blob/b269cde5231915d2930eefba55b706012843a155/src/tray.js#L305-L307

```js
    menu.getMenuItemById('checkForUpdates').enabled = !isUpdating
    menu.getMenuItemById('checkForUpdates').visible = !isUpdating
    menu.getMenuItemById('checkingForUpdates').visible = isUpdating
```